### DISCUSSION TITLE
Show all changed files in diff mode?

### DISCUSSION THREAD

**Discussion by ben-davis** [jesseduffield/lazygit#1472] [Q&A]
When using diff mode to diff two different branches/commits, is it possible to view the diff based on files changed? Just like we can when pressing `enter` on a commit.

**Comment by jesseduffield** [comment]
For this I typically press enter on the selected branch and then enter again on the HEAD commit for that branch. That will show you the difference in files between the two diffed items

**Reply by hacker-DOM** [reply]
Hey guys! Coming here after two years, I was wondering if this is now/still possible to view diffs of two commits (for simplicity let's say the HEAD and a historical commit).

I wasn't able to get this guide to work. Assuming I'm on the HEAD of branch `main`, if I go to the commit I want, my options are:

- space - checkout - not what I want
- enter - view that commit - not what I want either

Alternatively, I can click W and I get:
- diff [commit hash my cursor is on]  -  weirdly enough, this seems to produce the output `git diff [commit hash my cursor is on] [commit hash my cursor is on]` (there is a hint for this on the bottom right of the page:

<img width="1318" alt="image" src="https://github.com/jesseduffield/lazygit/assets/18601956/8d6cbbee-4e62-4930-bbe6-8d1726109653">

- enter ref to diff - this seems to compare the commit my cursor is on with the a user-specified commit. But the file tree doesn't update to allow me to view file by file, so I just have to scroll the long diff 🙁 

**Suggestion**

In `W` mode (when we have "showing output for ..." on lower right), we could update the file tree to that diff to allow people to compare commits file by file 🙂 WDYT?

**Reply by stefanhaller** [reply]
First, how to use `W`: navigate to the first commit, hit `W` and enter (this will initially show an empty diff, because it's diffing the current commit against itself). Now the diff dynamically updates as you move around; it always shows the diff between the commit you were on when you entered the mode, and the commit that you are on now.

As for seeing the individual file changes: The files panel is for showing your working copy changes, it would be the wrong place to show the diff you want to see. But you can simply hit enter when you are in diff mode, it will show you the changed files of the diff between whatever two commits you have chosen.

**Reply by hacker-DOM** [reply]
Thanks 🙏  Now let's get support for `difftastic` and lazygit's going to be be OP 🙂
